src/Microsoft.Health.Fhir.Api/Controllers/ExportController.cs

```diff
@@ -59,9 +75,24 @@ public class ExportController : Controller
         [Route(KnownRoutes.Export)]
         [ValidateExportHeadersFilter]
         [AuditEventType(AuditEventSubType.Export)]
-        public IActionResult Export()
+        public async Task<IActionResult> Export()
         {
-            return CheckIfExportIsEnabledAndRespond();
+            if (!_exportConfig.Enabled)
+            {
+                throw new RequestNotValidException(string.Format(Resources.UnsupportedOperation, OperationsConstants.Export));
+            }
+
+            CreateExportResponse response = await _mediator.ExportAsync(_fhirRequestContextAccessor.FhirRequestContext.Uri);
+
```

Comment:
This logic should go into core. You will want the logic in controller to be as thin as possible. (If someone wants to take our code and host it using another technology, they should be able to do it with minimum code change.)